edge_detection/driver.py

```diff
@@ -0,0 +1,248 @@
+#  Copyright (c) 2024 Mira Geoscience Ltd.
+#
+#  This file is part of geoapps.
+#
+#  geoapps is distributed under the terms and conditions of the MIT License
+#  (see LICENSE file at the root of this source code package).
+
+
+from __future__ import annotations
+
+import sys
+
+import numpy as np
+from geoapps_utils.driver.driver import BaseDriver
+from geoh5py.data import FloatData
+from geoh5py.groups import ContainerGroup, Group
+from geoh5py.objects import Curve, Grid2D, ObjectBase
+from geoh5py.ui_json import InputFile
+from skimage.feature import canny
+from skimage.transform import probabilistic_hough_line
+
+from .params import ApplicationParameters, DetectionParameters
+
+
+class EdgeDetectionDriver(BaseDriver):
+    """
+    Driver for the edge detection application.
+
+    :param parameters: Application parameters.
+    """
+
+    def __init__(self, parameters: ApplicationParameters | InputFile):
+        if isinstance(parameters, InputFile):
+            if not isinstance(parameters.data, dict):
+                raise TypeError("InputFile must have a 'data' dictionary.")
+
+            parameters = ApplicationParameters.parse_input_data(parameters.data)
```

Comment:
I would just check if it's an InputFile and dump directly to parse_input_data since it's already checking if it's a dict or InputFile object.

Reply:
Let parse_input_data do any validation like the one starting on line 34.  However the line 34 validation seems gratuitous to me - what scenario would we get an input file with .data that isn't a dictionary?

Reply:
That's just how we typed `InputFile.data: dict | None`

Reply:
ok, I guess it makes sense that we need different behaviours when InputFile.data is empty: raise exception at the driver level, but silently fail to load data at the params level.